src/main/kotlin/com/viartemev/thewhiterabbit/channel/TxChannel.kt

```diff
@@ -0,0 +1,85 @@
+package com.viartemev.thewhiterabbit.channel
+
+import com.rabbitmq.client.Channel
+import com.viartemev.thewhiterabbit.consumer.ConfirmConsumer
+import com.viartemev.thewhiterabbit.publisher.OutboundMessage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mu.KotlinLogging
+import java.util.concurrent.atomic.AtomicBoolean
+
+private val logger = KotlinLogging.logger {}
+
+@DslMarker
+annotation class TxChannelDslMarker
+
+@TxChannelDslMarker
+open class TxChannel internal constructor(private val channel: Channel) : Channel by channel {
```

Comment:
Should it be uncloseable?

Reply:
Actually it is, since client code works with `Channels.UnclosableTxChannel` wrapper. BTW this creates sort of an issue on tx rollback.

Reply:
What the issue?
Sorry for being absent, but I had bunch of conferences and all the staff